### PR TITLE
Fix missing profile docs

### LINKED ISSUE #4843: some profile are missing in the ipfs init command documentation
#### Version information:
go-ipfs version: 0.4.14-rc2-
Repo version: 6
System version: amd64/linux
Golang version: go1.10rc2

#### Type:
Bug

#### Description:
`ipfs init --help` list only 2 profiles (server and test) while more actually exist (https://github.com/ipfs/go-ipfs/pull/4195/files#diff-d39a8a737bc5c13f36e41aa92a0c7602). Ideally, this documentation would be generated to be always up to date.

Additionally, profile coule be listed in `ipfs config profile apply` as well.

### PR DESCRIPTION
**Pull request by magik6k**
Fixes #4843

### CODE REVIEW

File: repo/config/profile.go [hunk 1]
```diff
@@ -24,54 +33,96 @@ var defaultServerFilters = []string{
...
+var Profiles = map[string]Profile{
+	"server": {
+		Description: `Disables local host discovery, recommended when
+running IPFS on machines with public IPv4 addresses.`,
+
+		Transform: func(c *Config) error {
+			c.Addresses.NoAnnounce = appendSingle(c.Addresses.NoAnnounce, defaultServerFilters)
+			c.Swarm.AddrFilters = appendSingle(c.Swarm.AddrFilters, defaultServerFilters)
+			c.Discovery.MDNS.Enabled = false
+			return nil
```
Comment (Kubuxu) [inline repo/config/profile.go]: Quick Q: shouldn't server profile also disable UPnP/NAT propagation?

File: repo/config/profile.go [hunk 2]
```diff
@@ -24,54 +33,96 @@ var defaultServerFilters = []string{
...
-			},
-		}
-		return nil
+	"badgerds": {
+		Description: `Replaces default datastore configuration with experimental
+badger datastore.
+
+If you apply this profile after ipfs init, you will need
+to convert your datastore to the new configuration.
+You can do this using ipfs-ds-convert.
```
Comment (VictorBjelkholm) [inline repo/config/profile.go]: Can we put a usage example of ipfs-ds-convert as well? Or maybe a link to the readme of ipfs-ds-convert is enough

File: repo/config/profile.go [hunk 3]
```diff
@@ -24,54 +33,96 @@ var defaultServerFilters = []string{
...
-		return nil
+var Profiles = map[string]Profile{
+	"server": {
+		Description: `Disables local host discovery, recommended when
+running IPFS on machines with public IPv4 addresses.`,
+
+		Transform: func(c *Config) error {
+			c.Addresses.NoAnnounce = appendSingle(c.Addresses.NoAnnounce, defaultServerFilters)
+			c.Swarm.AddrFilters = appendSingle(c.Swarm.AddrFilters, defaultServerFilters)
+			c.Discovery.MDNS.Enabled = false
```
Comment (VictorBjelkholm) [inline repo/config/profile.go]: Why is MDNS disabled? Isn't the problem that we spam nearby ip addresses, not that we announce mdns services?

Comment (magik6k) [reply repo/config/profile.go]: IIRC Some providers don't like use of multicast addresses.

File: repo/config/profile.go (hunk 1 above)
Comment (whyrusleeping) [reply repo/config/profile.go]: Yeah, it should

File: repo/config/profile.go (hunk 2 above)
Comment (magik6k) [reply repo/config/profile.go]: Done

**Comment by whyrusleeping** [comment]
Thanks @magik6k  :)

**Pull request merged**